default/python/AI/ProductionAI.py

```diff
@@ -96,15 +97,18 @@ def cur_best_military_design_rating():
     if current_turn in _best_military_design_rating_cache:
         return _best_military_design_rating_cache[current_turn]
     priority = PriorityType.PRODUCTION_MILITARY
-    if priority in _design_cache:
-        if _design_cache[priority] and _design_cache[priority][0]:
-            rating, pid, design_id, cost = _design_cache[priority][0]
-            _best_military_design_rating_cache[current_turn] = rating
-            return max(rating, 0.001)
-        else:
-            return 0.001
-    else:
-        return 0.001
+    if _design_cache.get(priority, None) and _design_cache[priority][0]:
+        # the rating provided by the ShipDesigner does not
+        # reflect the rating used in threat considerations
+        # but takes additional factors (such as cost) into
+        # account. Therefore, we want to calculate the actual
+        # rating of the design as provided by CombatRatingsAI.
+        _, _, _, _, stats = _design_cache[priority][0]
+        # TODO: Should this consider enemy stats?
+        rating = CombatRatingsAI.ShipCombatStats(stats=stats.convert_to_combat_stats()).get_rating()
```

Comment:
Hmm, thinking about this a bit more, the mechanics of just how we'd want to have the default be an up-to-date value for foAI.foAIstate.get_standard_enemy(), while still allowing for None to be an acceptable value, might take a  bit of thinking/talking; it's far simpler to just let the default be None as you suggest and then we just change the current calls in PriorityAI and in MilitaryAI so that they specify enemy_stats=foAI.foAIstate.get_standard_enemy() when they call into cur_best_military_design_rating